requirements/default.txt

```diff
@@ -6,3 +6,4 @@ pillow>=5.4.0,!=7.1.0,!=7.1.1
 imageio>=2.4.1
 tifffile>=2019.7.26
 PyWavelets>=1.1.1
+scikit-learn(>=0.18.1)
```

Comment:
Pretty sure this should just be:
`scikit-learn>=0.18.1`

but it should be in `requirements/optional.txt` instead of this file

I think the package name must match the name of the PyPI package and the `-` in the name should not be a problem (for example we have a package `sphinx-gallery` in our requirements/docs.txt)

Reply:
I originally had it like that, but it was failing with the same error, so I tried this.

I'll put it in `requirements/optional.txt` anyway.

Reply:
@grlee77  Do you have any suggestions on how to prevent CI from failing since sklearn is now an optional import? Even if I catch and reraise the import error, this won't help with CI failing.

Reply:
@grlee77 Done. I moved to optional dependencies and added the pytest.importorskip

Reply:
Done